Intersect.Server.Core/Entities/Player.cs

```diff
@@ -2125,11 +2125,45 @@ public bool CanChangeToInstanceType(MapInstanceType? instanceType, bool fromLogi
                 case MapInstanceType.Shared:
                     if (fromLogin)
                     {
-                        isValid = false;
+                        // Check to see if the instance still exists & is populated
+                        if (!InstanceProcessor.TryGetInstanceController(SharedMapInstanceId, out var sharedController) || sharedController.Players.Count == 0)
+                        {
+                            return false;
+                        }
+
+                        // If it does, get the party leader from it...
+                        var partyMember = sharedController.Players.FirstOrDefault();
+                        if (partyMember == default)
```

Comment:
What happens if _you_ were the party leader? I actually have no idea what happens to a 3+ person party if the lead disconnects

Reply:
The leadership goes to the next in line of the array.